### ISSUE TITLE
Misleading error message related to externs and NTI

### ISSUE THREAD

**Issue by bkputnam** [google/closure-compiler#2581]
When I include my externs files in the wrong order I get a misleading error message. The 3 examples below show what my code should have been (externs in the right order), what I think I should have gotten as an error message, and what I actually got.

For reference, my build setup uses `--new_type_inf` and considers warnings to be errors. No warning is presented without the `--new-type-inf` flag.

1: The correct version. This is what my code should have been, and it compiles correctly with no errors. Note that `gapi.client.Foo` is referenced before it is defined and this is not an issue.

```javascript
var gapi = {};

gapi.client = {};

/** @type {!gapi.client.Foo} */
gapi.client.foo_instance;


/**
* @typedef {{
*   v1: {
*     items: function(!Object=, (!Object|string)=)
*   }
* }}
*/
gapi.client.Foo;
```

2: If I completely forget the namespaces `gapi` and `gapi.client` I get a useful error message:

```javascript
/** @type {!gapi.client.Foo} */
gapi.client.foo_instance;


/**
* @typedef {{
*   v1: {
*     items: function(!Object=, (!Object|string)=)
*   }
* }}
*/
gapi.client.Foo;
```

The error message:

```
extern:2: WARNING - name gapi is not defined in the externs.
gapi.client.foo_instance;
^^^^

extern:12: WARNING - name gapi is not defined in the externs.
gapi.client.Foo;
^^^^

extern:2: ERROR - variable gapi is undeclared
gapi.client.foo_instance;
^^^^

1 error(s), 2 warning(s)
```

3: The misleading error message. If I include `gapi` and `gapi.client` but I put their definitions after they're needed, I get this error message. (I don't think this makes a difference, but in reality `gapi` and `gapi.client` live in a different file than `gapi.client.Foo` and `gapi.client.foo_instance` and their files were included in the wrong order). The error message below lead me to believe that I had a reference-before-declaration error even though, per example 1, reference-before-declaration is supported in this scenario.

```javascript
/** @type {!gapi.client.Foo} */
gapi.client.foo_instance;


/**
* @typedef {{
*   v1: {
*     items: function(!Object=, (!Object|string)=)
*   }
* }}
*/
gapi.client.Foo;

var gapi = {};

gapi.client = {};
```

Error message:

```
extern:1: WARNING - Type annotation references non-existent type gapi.client.Foo.
/** @type {!gapi.client.Foo} */
            ^^^^^^^^^^^^^^^

0 error(s), 1 warning(s)
```

**Comment by brad4d** [comment]
Improving the error message here would definitely be a good thing, but I don't think anyone on the team has time to spend on it just now, so I'm marking this "help wanted".

**Comment by dimvar** [comment]
Actually, I'll take a look at this sometime in the next few days; it might be a simple fix.

**Comment by dimvar** [comment]
This should be fixed by commit 88a46ccbb5a99eedb1658b76394ae81a4c6f8363
